events/git-and-github.md

```diff
@@ -0,0 +1,129 @@
+
+   Again it is a good point stop and summarise branching. Pause for short period of time. Let audience ask their queries. 
+
+9. **Introduce Github**
+
+   Now once the audience are comfortable with basics of Git. It is the right time to introduce Github. Tell them, how Github is important for software development. Give an analogy of people writing a shared google doc, and explain collaborative software development better.
+
+10. **Push and Pull**
+
+   This is best explained via demo. First show them how to create SSH keys and how to add the generated keys to Github. Show them how to make a new repository    on Github. 
+   
+   Show them how to push the local project to the Github repo using `git push git@github.com:USERNAME/REPONAME.git master` . Make a README on repo on Github. Show them how the changes on Github repo can be obtained locally via `git pull git@github.com:USERNAME/REPONAME.git master`.
+
+   Tell them typing the URL is a pain every time, and use this as an opportunity to introduce the concept of remote variables. Add a remote by name `origin`  using `git remote add origin` and show how `git push origin master` does the same job. 
+
+11. **Sending a Pull Request**
+
+    This is also best explained via a demo. It is a good idea to have a Github Repository made from KOSS account for the Pull Request demo. Make a small issue like a  "typo in `printf` function" which anyone can solve. Follow up the whole process from checking the issues, comment on issue you want to work, Forking the repo, Cloning it, Making a new branch, Push it, Sending a Pull Request with proper description. 
+
+    While demonstrating each and every step explain clearly, why you are doing this. For example, tell them that you are forking the repo to have your own copy of the codebase. Similarly, make your explanation in such a way that steps have a logical sequence.
+
+    After the demo, show them pictorially all the steps from Forking to Sending a PR. Ask the audience to send a PR. Tell them that sending the PR is the most important take away of workshop. TAs in the workshop can provide individual assistance. If needed, you can show the demo again.
+
+    Finally merge a PR which contains a well written description to complete the process. 
+
+12. **Conclusion**
+
+    - Tell them to contribute to beginner friendly issues at metakgp or KOSS repositories on Github. Share the Github Links of both the Github Organizations. Pointing them to some beginner friendly issues would be great.
+    - Tell them about Github Student pack and its benefits like Domains, Free Server and so on.
+    - Tell them about Github Pages, and tell them howeasily a beautiful blog can be setup within minutes.
+    - Inform about Hacktoberfest and tell them about the T-shirt and Stickers
+
+    - Tell them about starring a repo and getting the notifications.
+    - Ask them to star `kossiitkgp/slides` as important links like slides, resources will be put here
```

Comment:
One can use the releases to push newer changes as notifications. The release follower will not have to manually check the starred repo for new changes. 

I think Following releases is a much better way to track a project/resource than starring it.

Reply:
Amazing
Changed the docs! Thanks for mentioning it. 

From next time onwards, we will push releases to notify changes. This can be included in `README` of slides repo too!